test/Microsoft.Extensions.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -219,5 +241,20 @@ public interface IBar { }
         public class Bar1 : IBar { }
 
         public class Bar2 : IBar { }
+
+        private class RequiredServiceAwareServiceProvider : IServiceProvider, IRequiredServiceAware
+        {
+            public const string ErrorMessageFormat = "Required service {0} was not found.";
+
+            public object GetRequiredService(Type serviceType)
+            {
+                throw new InvalidOperationException(string.Format(ErrorMessageFormat, serviceType.Name));
+            }
+
+            public object GetService(Type serviceType)
+            {
+                return false;
```

Comment:
You could `throw new NotSupportedException` here since it's never going to get called.